broker/src/main/java/io/camunda/zeebe/broker/exporter/repo/ExporterRepository.java

```diff
@@ -88,7 +88,15 @@ private void validate(final ExporterDescriptor descriptor) throws ExporterLoadEx
     try {
       final Exporter instance = descriptor.newInstance();
       final ExporterContext context = new ExporterContext(LOG, descriptor.getConfiguration());
-      instance.configure(context);
+
+      final var currentThread = Thread.currentThread();
+      final var previousClassLoader = currentThread.getContextClassLoader();
+      try {
+        currentThread.setContextClassLoader(instance.getClass().getClassLoader());
+        instance.configure(context);
+      } finally {
+        currentThread.setContextClassLoader(previousClassLoader);
+      }
```

Comment:
Would it be possible to reuse the ExporterContainer's checkedRunWithClassLoader? It seems to be the same logic and `() -> instance.configure(context)` would fit the CheckedRunnable interface. If it's possible, this would fit DRY. If it doesn't, I think it might be good to add a comment here why this is necessary (perhaps even referring to the ExporterContainer)

Reply:
I considered it briefly, but what component would you use to share this code? It's too special purpose I think for a general utility, and it only has one small case of duplication.

It actually made me think that the approach to validation is possibly incorrect - maybe we should instead build the container once at startup, and just keep it around, calling open/close? It does put more burden on the exporter authors to clean up their resources and ensure the same instance can be reused though, so it has some non negligible downsides.

Anyhow, if you have some idea of where it would fit to reuse/share this code, I'm open to it. I just didn't really see it.

Reply:
I don't really see it either, I was thinking about just making `checkedRunWithClassLoader` in `ExporterContainer` public and static. You'd need to pass in the exporter instance to use that way. Not sure if that's nice to use. But like I said, it's also fine to just add a comment why this is necessary and possibly referring to ExporterContainer.

Reply:
I thought about it some more, and in theory this is thread specific code - you could do something like:

```java
public final class ThreadUtil {
  public static void runCheckedWithClassLoader(
      final CheckedRunnable runnable, final ClassLoader classLoader) throws Exception {
    final var currentThread = Thread.currentThread();
    final var contextClassLoader = currentThread.getContextClassLoader();

    try {
      currentThread.setContextClassLoader(classLoader);
      runnable.run();
    } finally {
      currentThread.setContextClassLoader(contextClassLoader);
    }
  }
}
```

That's not too crazy - however in our code base it's still only used for really on purpose, side loading external exporters, so I'm unsure about moving it into something like `zeebe-util`. We could have it as a utility in the exporter package maybe?

Reply:
I like the idea to keep this in the exporter package. There's a big chance we remove this in the future, when we rework exporters.

Reply:
> when we rework exporters

Such optimism :smile: